File: packages/dk-data-client/python/dk_data_client/modules/providers.py
"""Healthcare provider read surface.

Reads from `hcs_silver.providers` via the FastAPI resolve wrapper.
"""

from __future__ import annotations

from typing import Any

from dk_data_client.modules._base import ModuleBase


class ProvidersModule(ModuleBase):
    async def resolve(self, npi_or_name: str, *, hint: str | None = None) -> dict[str, Any]:
        body: dict[str, Any] = {"name_or_id": npi_or_name}
        if hint:
            body["hint"] = hint
        return await self._call(
            method="providers.resolve",
            path="/data-platform/providers/resolve",
            args={"npi_or_name": npi_or_name, "hint": hint},
            http_method="POST",
            json_body=body,
        )

    async def get(self, provider_id: str) -> dict[str, Any]:
        params = {"provider_id": f"eq.{provider_id}", "limit": "1"}
        rows = await self._call(
            method="providers.get",
            path="/providers",
            args={"id": provider_id},
            params=params,
        )
        return dict(rows[0]) if isinstance(rows, list) and rows else {}

File: packages/dk-data-client/typescript/src/modules/providers.ts
/** Healthcare provider read surface. */

import { ModuleBase } from "./_base.js";

export class ProvidersModule extends ModuleBase {
  async resolve(
    npiOrName: string,
    opts: { hint?: string } = {},
  ): Promise<Record<string, unknown>> {
    const body: Record<string, unknown> = { name_or_id: npiOrName };
    if (opts.hint) body.hint = opts.hint;
    return (await this.call({
      method: "providers.resolve",
      path: "/data-platform/providers/resolve",
      args: { npiOrName, hint: opts.hint },
      httpMethod: "POST",
      jsonBody: body,
    })) as Record<string, unknown>;
  }

  async get(providerId: string): Promise<Record<string, unknown>> {
    const rows = (await this.call({
      method: "providers.get",
      path: "/providers",
      args: { id: providerId },
      params: { provider_id: `eq.${providerId}`, limit: "1" },
    })) as Array<Record<string, unknown>>;
    return Array.isArray(rows) && rows.length > 0 ? rows[0]! : {};
  }
}
